### PR TITLE
Update instructions on how to access a Samba export (#110).

### LINKED ISSUE #110: Add how-to for Samba access from different clients
Window, Linux and Mac.

### LINKED ISSUE #343: IP address-based access format for windows is incorrect in file sharing documentation
In the documentation here:
https://rockstor.com/docs/interface/storage/file_sharing/samba_ops.html#how-to-access-a-share

The ip based access needs to contain 2 leading backslashes instead of the single one depicted in the documentation.
So, it should be `\\192.168.1.21` instead.

Reference to forum thread:
https://forum.rockstor.com/t/importing-loading-files-from-usb-fat-structured/8132/10?u=hooverdan

### PR DESCRIPTION
**Pull request by FroggyFlox**
Fixes #110 
Fixes #343 
@phillxnet, submitting as "draft" for now waiting for double-checking in buildbot.

### This pull request's proposal
See #110 for detailed history and explanations for changes. Briefly, this pull request (PR) proposes to centralize and harmonize our current documentation on file sharing and how to access Rockstor shares exported using the Samba protocol. This thus includes updated screenshots and description on how to create a Samba export, view/edit/delete a Samba export, as well as detailed graphical instructions on how to access a Samba export from the three main clients (Linux, Windows, and macOS). Note that the instructions for Windows are more developed than for the other two clients as this is likely the most used case.

### Commits summary
- Detail Samba service configuration.
- Add step-by-step graphical instuctions on how to access a Samba export from
Windows manually by entering the share's path
- Escape backslash when printing Windows-style network path (#343)
- Add graphical instructions on accessing a samba export from a Windows machine
using the "add a network location" method
- Add instructions on how to access a Samba export using the "Map a network
drive" option
- Add instructions on how to access a Samba export from Dolphin
- Update graphical instructions for macOS
- Move placeholder on how to access NFS shares to nfs_ops.rst
- Add general recommendations on the choice of file sharing protocol
- Update screenshot for Samba exports table and harmonize format for
View/Edit/Delete a Samba export section.

### Checklist
- [x] With the proposed changes no Sphinx errors or warnings are generated.
- [x] I have added my name to the AUTHORS file, if required (descending alphabetical order).

### CODE REVIEW

File: interface/storage/file_sharing/samba_ops.rst [hunk 1]
```diff
@@ -1,138 +1,339 @@
...
+
+In our *Photos* share example, this corresponds to:
+
+.. image:: /images/interface/storage/file_sharing/samba_ops/samba_access_win_network_location_wizard_path.png
+   :align: center
+
+
+Click *Next*. When asked for credentials, use the corresponding username and
+password. As we set up the *Photos* share to be accessed by the user named
+*media_user*, we will enter its login and password:
```
Comment (phillxnet) [inline interface/storage/file_sharing/samba_ops.rst:244]:
```suggestion
*media_user*, we will enter their login name and password:
```

File: interface/storage/file_sharing/samba_ops.rst [hunk 2]
```diff
@@ -1,138 +1,339 @@
...
+
+.. note::
+
+  The domain name is defaulted to your machine name. Enter :code:`DOMAINNAME\username`
+  in the username field to use an alternative domain name (such as an AD
+  domain, for instance).
+
+Check *Remember my credentials*, and click OK.
+
+You will then be asked to enter the name of your choice for this share. This
```
Comment (phillxnet) [inline interface/storage/file_sharing/samba_ops.rst:258]:
```suggestion
You will then be asked to enter a name of your choice for this share. This
```

File: interface/storage/file_sharing/samba_ops.rst [hunk 3]
```diff
@@ -1,138 +1,339 @@
...
+"""""""""""""""""""
 
-1. On your Mac, open the Finder, and select *Go* -> *Connect to Server*
+In the Windows Explorer menu, click *Map network drive*. This will open a
+dedicated wizard. When asked for the address of the network location, enter the
+address of the Rockstor share using the following format:
 
-2. In the Server address field, type in the address of the shared folder as
-   shown below (with the ip address of your Rockstor appliance)
+    \\\\<rockstor-appliance-ip>\\<share-name>
```
Comment (phillxnet) [inline interface/storage/file_sharing/samba_ops.rst:279]: Same again here re the rendering. I get:

>     \<rockstor-appliance-ip>\<share-name>

File: interface/storage/file_sharing/samba_ops.rst [hunk 4]
```diff
@@ -1,138 +1,339 @@
...
 
-3. Click Connect. The Finder will connect to the Rockstor appliance, and will
-   popup an authentication dialog.
+.. image:: /images/interface/storage/file_sharing/samba_ops/samba_access_win_map_path.png
+   :align: center
+
+
+Click *Next*. When asked for credentials, use the corresponding username and
+password. As we set up the *Photos* share to be accessed by the user named
+*media_user*, we will enter its login and password:
```
Comment (phillxnet) [inline interface/storage/file_sharing/samba_ops.rst:289]:
```suggestion
*media_user*, we will enter their login name and password:
```

Comment (phillxnet) [reply interface/storage/file_sharing/samba_ops.rst:289]: To be consistent with the other similar suggestions.

File: interface/system/services.rst [hunk 5]
```diff
@@ -32,13 +32,45 @@ Rockstor uses Linux NFS server to support exporting Shares to remote clients
...
+- **Workgroup** is the only *required* setting. This should be set to the
+  Windows NT domain or the workgroup name. Unless changed, the default domain
+  name for a Windows machine is :code:`WORKGROUP`.
+- The **Custom global configuration** (*optional*) box allows you to enter any
+  custom samba setting needed to customize the Samba service to your needs. The
+  possible options are thus all those supported by Samba in its :code:`smb.conf`
+  file. See the `Samba wiki <https://www.samba.org/samba/docs/current/man-html/smb.conf.5.html>`_
+  for a full list of options and their documentation.
+
+When ready, the Samba service, press the *Submit* button to save the
```
Comment (phillxnet) [inline interface/system/services.rst:67]:
```suggestion
When ready, press the *Submit* button to save the
```

Comment (phillxnet) [reply interface/system/services.rst:67]: I'm assuming this was a typo, so just shortening for now.

File: interface/system/services.rst [hunk 6]
```diff
@@ -32,13 +32,45 @@ Rockstor uses Linux NFS server to support exporting Shares to remote clients
...
+configuration changes.
+
+Once you are done configuring it, the Samba service can now be turned ON by
+toggling the ON/OFF slider located to its right.
```
Comment (phillxnet) [inline interface/system/services.rst:71]:
```suggestion
toggling the ON/OFF switch located to its right.
```

**Review by phillxnet — CHANGES_REQUESTED** [review]
@FroggyFlox This is a major improvement. Well done.

I've popped in a few suggestions given your extreme efforts here.

Thanks again for this and the only thing left pre-merge is to figure out the formatting to display in production the "//" bit and to move this pr from draft as I can't merge a draft.

Apologies if I've jumped in too soon on this. I saw the needs review tag and acted on that.

**Comment by phillxnet** [comment]
@FroggyFlox Re:
> submitting as "draft" for now waiting for double-checking in buildbot.

All looked good to me rendering wise except the indicated in review "\\" bit.

File: howtos/clients.rst [hunk 7]
```diff
@@ -4,70 +4,25 @@ Accessing Shares from Clients
...
+share. For instance, while **Samba** exports can be accessed by Linux, Windows,
+and macOS clients alike, **NFS** exports cannot be easily accessed from Windows
+clients. In addition, each protocol varies in its performance and ease of
+access. As a result, the decision of using a protocol over another for a given
+share ultimately depends on your exact needs. We can however make the general
+recommendations below:
+
+- If your share needs to be accessed by at least one Windows clients, use
+  **Samba**. See our :ref:`samba_export` section for details.
+- If your share needs to be accessed by Linux clients *only*, use **NFS**.
```
[the remark this answers is not in the document]
Comment (FroggyFlox) [reply howtos/clients.rst:22]: Very good points and I agree with you. I really like your change in wording as it's a very elegant way to convey this.
Thanks!

File: interface/storage/file_sharing/samba_ops.rst (hunk 3 above)
Comment (FroggyFlox) [reply interface/storage/file_sharing/samba_ops.rst:279]: Curious... It looked correct when testing using `python -m http.server ...`:
image

I suppose you checked under apache2?

**Comment by FroggyFlox** [comment]
Thanks a lot for finding all these errors, @phillxnet !

See my comment re the double backslash... It rendered ok for me under `python -m http.server` so I suppose it's another trap here... Maybe we should simply use a code block here?

File: interface/storage/file_sharing/samba_ops.rst (hunk 3 above)
Comment (phillxnet) [reply interface/storage/file_sharing/samba_ops.rst:279]: Yes, via the buildbot "PUBLISH-Docs" Builder with local target and this pr number entered.

**Comment by phillxnet** [comment]
Just double checking the rendering again.

**Comment by phillxnet** [comment]
@FroggyFlox Just checked again and yes we have, via a proper html server:
double-backslash-rendering

Also some of my  "enter its login and password:" -> "their login name and password:" suggestions have been resolved but not committed. Is that on purpose. If so dandy. But just wanted to know if this was an intentional resolve without commit?

Really nice upgrade to this rather neglected doc section.

**Comment by FroggyFlox** [comment]
> Also some of my "enter its login and password:" -> "their login name and password:" suggestions have been resolved but not committed. Is that on purpose. If so dandy. But just wanted to know if this was an intentional resolve without commit?

Thanks for bringing this up; I didn't notice those... they were somehow collapsed as "hidden" so I had missed them... Hopefully I committed everything correctly this time.

I'll try a code block here and see how it looks.

**Comment by FroggyFlox** [comment]
@phillxnet , can you check the rendering of the double backslash post https://github.com/rockstor/rockstor-doc/pull/351/commits/6c70567b261df63a18f401f01899250472480fbe at your convenience?

Sorry, I don't have access to my buildbot at the moment :\

**Comment by phillxnet** [comment]
@FroggyFlox Sorry got called away for a bit.
> can you check the rendering of the double backslash post [6c70567](https://github.com/rockstor/rockstor-doc/commit/6c70567b261df63a18f401f01899250472480fbe)

That's learned it:
double-backslash-rendering-code-block

If you fancy doing the same on the other two I can merge and publish if you are happy with that.

**Comment by phillxnet** [comment]
Just noting the transient failure of our link check earlier:
```
(interface/storage/file_sharing/apple_filing_protocol: line   25) ok        https://support.apple.com/en-us/HT202784
412
(interface/storage/file_sharing/apple_filing_protocol: line   13) broken    https://support.apple.com/en-gb/HT204445 - HTTPSConnectionPool(host='support.apple.com', port=443): Read timed out. (read timeout=20)
```

**Comment by FroggyFlox** [comment]
Thanks a lot for double checking (and sorry for the extra work :(

> If you fancy doing the same on the other two I can merge and publish if you are happy with that.

Done... hopefully I didn't forget anything else this time!

**Comment by phillxnet** [comment]
@FroggyFlox Thanks. This is a really nice addition. I've now checked the merged result locally and published to production.

Well done.

**Pull request merged**